File: java/src/main/java/com/ollang/sdk/resources/Content.java
package com.ollang.sdk.resources;

import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.ollang.sdk.OllangClient;
import java.util.ArrayList;
import java.util.Collections;
import java.util.LinkedHashMap;
import java.util.List;
import java.util.Map;

/** Imports and exports translation units held in your content database. */
public class Content {

  private final OllangClient client;

  public Content(OllangClient client) {
    this.client = client;
  }

  /**
   * Imports translation units.
   *
   * <p>Each entry of {@code translations} needs {@code sourceText} and {@code targetText}, and
   * may carry {@code elementId} and {@code type} (defaults to {@code text}).
   */
  public JsonElement importContent(String targetLanguage, JsonArray translations) {
    JsonObject body = new JsonObject();
    body.addProperty("targetLanguage", targetLanguage);
    body.add("translations", translations);
    return client.post("/integration/content/import", body);
  }

  /** Exports content translations as JSON. */
  public JsonElement export() {
    return export(null);
  }

  /**
   * Exports content translations as JSON, filtered by language, tag or order.
   *
   * <p>List filters are sent as repeated bracket-suffixed query parameters ({@code
   * targetLanguages[]=fr&targetLanguages[]=de}), which is the encoding the API expects.
   */
  public JsonElement export(ExportOptions options) {
    return client.getMulti(
        "/integration/content/export", options == null ? null : options.toQueryParams());
  }

  /** Filters for {@link #export(ExportOptions)}. */
  public static class ExportOptions {

    private final Map<String, List<String>> params = new LinkedHashMap<>();

    public ExportOptions targetLanguage(String targetLanguage) {
      params.put("targetLanguage", Collections.singletonList(targetLanguage));
      return this;
    }

    public ExportOptions targetLanguages(List<String> targetLanguages) {
      params.put("targetLanguages[]", new ArrayList<>(targetLanguages));
      return this;
    }

    public ExportOptions tag(String tag) {
      params.put("tag", Collections.singletonList(tag));
      return this;
    }

    public ExportOptions tags(List<String> tags) {
      params.put("tags[]", new ArrayList<>(tags));
      return this;
    }

    public ExportOptions orderIds(List<String> orderIds) {
      params.put("orderIds[]", new ArrayList<>(orderIds));
      return this;
    }

    public Map<String, List<String>> toQueryParams() {
      return params;
    }
  }
}

File: python/src/ollang/resources/content.py
from typing import Any, Dict, List, Optional

from .._client import OllangClient


class Content:
    """Import and export translation units held in your content database."""

    def __init__(self, client: OllangClient):
        self._client = client

    def import_(
        self,
        target_language: str,
        translations: List[Dict[str, Any]],
    ) -> Dict[str, Any]:
        """Import translation units.

        Each entry of ``translations`` needs ``sourceText`` and ``targetText``,
        and may carry ``elementId`` and ``type`` (defaults to ``text``).

        Named ``import_`` because ``import`` is a Python keyword.
        """
        return self._client.post(
            "/integration/content/import",
            json={"targetLanguage": target_language, "translations": translations},
        )

    def export(
        self,
        target_language: Optional[str] = None,
        target_languages: Optional[List[str]] = None,
        tag: Optional[str] = None,
        tags: Optional[List[str]] = None,
        order_ids: Optional[List[str]] = None,
    ) -> Any:
        """Export content translations as JSON, filtered by language, tag or order.

        List filters are sent as repeated bracket-suffixed query parameters
        (``targetLanguages[]=fr&targetLanguages[]=de``), which is the encoding
        the API expects.
        """
        params: Dict[str, Any] = {}
        if target_language is not None:
            params["targetLanguage"] = target_language
        if target_languages is not None:
            params["targetLanguages[]"] = target_languages
        if tag is not None:
            params["tag"] = tag
        if tags is not None:
            params["tags[]"] = tags
        if order_ids is not None:
            params["orderIds[]"] = order_ids
        return self._client.get("/integration/content/export", params=params or None)
